omf-bifrost/overture-access-restriction-splitter/src/tests.rs

```diff
@@ -0,0 +1,351 @@
+#[cfg(test)]
+mod tests {
+    use crate::restriction_splitter::split_segments;
+    use overture_types::{ConnectorRef, Segment};
+    use geo_types::{LineString, Coord};
+
+    /// Helper function to build a segment with given geometry, access restrictions, and connectors
+    fn build_segment(geometry: &[Coord], access_restrictions: Vec<(f64, f64)>, connectors: Vec<ConnectorRef>) -> Segment {
+        let access_restrictions = access_restrictions.into_iter().map(|(start, end)| {
+            overture_types::properties::AccessRestriction {
+                access_type: "denied".to_string(),
+                between: Some((start, end)),
+                when: None,
+            }
+        }).collect();
+
+        Segment {
+            id: "id".to_string(),
+            geometry: LineString::from(geometry.to_vec()),
+            properties: overture_types::properties::SegmentProperties {
+                class: Some("residential".to_string()),
+                names: None,
+                access_restrictions: Some(access_restrictions),
+                speed_limits: None,
+                subtype: None,
+                surface: None,
+            },
+            connectors,
+        }
+    }
+
+    fn fuzzy_compare(a: f64, b: f64) -> bool {
+        let epsilon = 1e-4;
+        (a - b).abs() <= epsilon
+    }
+
+    fn fuzzy_compare_coords(c1: &Coord, c2: &Coord) -> bool {
+        fuzzy_compare(c1.x, c2.x) && fuzzy_compare(c1.y, c2.y)
+    }
+
+    #[test]
+    fn test_split_none() {
+        let segment = build_segment(
+            &[
+                geo_types::Coord { x: 0.0, y: 0.0 },
+                geo_types::Coord { x: 1.0, y: 0.0 },
+                geo_types::Coord { x: 2.0, y: 0.0 },
+                geo_types::Coord { x: 3.0, y: 0.0 },
+            ],
+            vec![],
+            vec![ConnectorRef{ id: "conn-1".to_string(), at: 0.5 }],
+        );
+        let mut segments = vec![segment];
+        split_segments(&mut segments);
+
+        // No splits should occur
+        assert_eq!(segments.len(), 1);
+
+        // Connector should be at the same spot
+        assert_eq!(segments[0].connectors.len(), 1);
+        assert!(fuzzy_compare(segments[0].connectors[0].at, 0.5));
+    }
+
+    #[test]
+    fn test_split_one() {
+        let segment = build_segment(
+            &[
+                geo_types::Coord { x: 0.0, y: 0.0 },
+                geo_types::Coord { x: 1.0, y: 0.0 },
+                geo_types::Coord { x: 2.0, y: 0.0 },
+                geo_types::Coord { x: 3.0, y: 0.0 },
+                geo_types::Coord { x: 4.0, y: 0.0 },
```

Comment:
Nice, thanks for adding tests. May want to add one or two more that have more complex "real-world" data, ideally directly from production data for the connector LR values to test projection mismatches